# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineCodeGen.cpp
# Commit: 5f6b9e2fc3bb
# Full Hash: 5f6b9e2fc3bb3c5fd74f9871399269a2665a5270
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2022-03-09 09:44:44
# Regressor Bug: 885514
# File Overlap Count: 1
# Description:
#   Bug 885514: Refactor try/finally to use goto r=jandem
#   
#   JSOp::Gosub is just a JSOp::Goto that lies about its effect on the stack. We can start simplifying try/finally by using a JSOp::Goto instead.
#   
#   Currently, to make the stack depth work out when returning from a finally, Gosub pretends to pop two values (the arguments to retsub), and Finally pretends to push them. This means that bytecode analysis can treat Gosub as if it falls through, and the stack depth of the jumptarget following the gosub will be correct. This patch uses a different approach: bytecode analysis marks the return address as reachable when we push it using JSOp::ResumeIndex. This means that the bytecode no longer has to lie about stack depth.
# ==============================================================================

diff -r b199b21ed5a3 -r 5f6b9e2fc3bb js/src/jit/BaselineCodeGen.cpp
--- a/js/src/jit/BaselineCodeGen.cpp	Tue Mar 08 21:00:30 2022 +0000
+++ b/js/src/jit/BaselineCodeGen.cpp	Tue Mar 08 21:10:21 2022 +0000
@@ -4628,10 +4628,6 @@
 
 template <typename Handler>
 bool BaselineCodeGen<Handler>::emit_Finally() {
-  // JSOp::Finally has a def count of 2, but these values are already on the
-  // stack (they're pushed by JSOp::Gosub). Update the compiler's stack state.
-  frame.incStackDepth(2);
-
   // To match the interpreter, emit an interrupt check at the start of the
   // finally block.
   return emitInterruptCheck();